### PR TITLE
Title in progress dialog

### LINKED ISSUE #15113: [Bug] Title missing in ProgressDialog
###### Reproduction Steps

1. Go to LoginActivity
2. Enter details 
3. Click Login

###### Expected Result
Dialog title is there as `Loading`

###### Actual Result
Dialog title is missing

https://github.com/ankidroid/Anki-Android/assets/65113071/d3fc1041-5bd6-4742-a694-3091aa619dab

###### Debug info

AnkiDroid Version = 2.17alpha14-debug (275a16f5513baccc417f081fa976ceae997cd86c)

Backend Version = 0.1.32-anki23.10.1 (23.10.1 fac9e0ee1436ba5ac3366c72dd9394a6e692b1cf)

Android Version = 13 (SDK 33)

ProductFlavor = amazon

Manufacturer = realme

Model = RMX3521

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 13; RMX3521 Build/RKQ1.211119.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/120.0.6099.144 Mobile Safari/537.36

ACRA UUID = c320a643-88c1-4346-8a07-57738b056962

Crash Reports Enabled = false

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by SanjaySargam**
## Purpose / Description
Title missing in ProgressDialog

## Fixes
* Fixes #15113

## Approach
set dialog title

## How Has This Been Tested?

Physical Device ( Realme 9 )

https://github.com/ankidroid/Anki-Android/assets/65113071/c64b46c9-b407-4ecd-98a2-eaeafd9fd759

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt [hunk 1]
```diff
@@ -161,7 +161,12 @@ fun MyAccount.handleNewLogin(username: String, password: String) {
     val endpoint = getEndpoint(this)
     launchCatchingTask {
         val auth = try {
-            withProgress({}, onCancel = ::cancelSync) {
+            withProgress(
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt:164]: Can't this be passed in as a parameter?

We don't have progress here

**Review by david-allison — COMMENTED** [review]
One question about using a named argument

File: AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt (hunk 1 above)
Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt:164]: I didn't understand what you meant. Will you please show how to pass this as a parameter

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt:164]: We don't have a notion of 'progress' here, it's a one-shot operation. Therefore we should likely have a `message` parameter

```diff
Subject: [PATCH] log to file
---
Index: AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt b/AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt
--- a/AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt	(revision 275a16f5513baccc417f081fa976ceae997cd86c)
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt	(date 1704028835990)
@@ -161,7 +161,7 @@
     val endpoint = getEndpoint(this)
     launchCatchingTask {
         val auth = try {
-            withProgress({}, onCancel = ::cancelSync) {
+            withProgress({}, message = getString(R.string.sign_in), onCancel = ::cancelSync) {
                 withCol {
                     syncLogin(username, password, endpoint)
                 }
Index: AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt b/AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt
--- a/AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt	(revision 275a16f5513baccc417f081fa976ceae997cd86c)
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt	(date 1704028767838)
@@ -244,6 +244,7 @@
 suspend fun <T> FragmentActivity.withProgress(
     extractProgress: ProgressContext.() -> Unit,
     onCancel: ((Backend) -> Unit)? = { it.setWantsAbort() },
+    message: String? = null,
     op: suspend () -> T
 ): T {
     val backend = CollectionManager.getBackend()
@@ -255,6 +256,8 @@
             null
         }
     ) { dialog ->
+        @Suppress("Deprecation") // ProgressDialog deprecation
+        message?.let { dialog.setMessage(it) }
         backend.withProgress(
             extractProgress = extractProgress,
             updateUi = { updateDialog(dialog) }

```

Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt:164]: Ok got it.
Thanks for explaining

Comment (SanjaySargam) [reply AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt:164]: @david-allison  I tried it, but it is not working. I mean title is not showing

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/Sync.kt:164]: sorry! let me look into things

**Review by david-allison — APPROVED** [review]
Was too time-consuming to fix my suggestion, sorry for the wasted time, looks good to go

**Comment by SanjaySargam** [comment]
> Was too time-consuming to fix my suggestion, sorry for the wasted time, looks good to go

No worries!

**Review by lukstbit — APPROVED** [review]

**Pull request merged**